app/src/main/java/com/example/weather/data/RepositoryImpl.java

```diff
@@ -66,19 +68,35 @@ public void updateWeather(WeatherData weatherData) {
     }
 
     @Override
-    public Observable<Float> getWeatherDataByCoord(String lat, String lon, String units) {
+    public Observable<Float> getCurrentWeatherDataByCoord(String lat, String lon, String units) {
         Observable<Float> floatObservable= ForecastApi.Instance.getForecastApi().getWeatherDataByCoord(lat, lon, units).map(
                 new Function<WeatherEntity, Float>() {
                     @Override
                     public Float apply(@NonNull WeatherEntity weatherEntity) throws Exception {
-                        Float tmp = weatherEntity.getMain().getTemp();
-                        return weatherEntity.getMain().getTemp();
+                        return weatherEntity.getCurrent().getTemp();
                     }
                 }
         );
         return floatObservable;
     }
 
+    @Override
+    public Observable<HashMap<String, Float>> getDailyWeatherDataByCoord(String lat, String lon, String units) {
+        Observable<HashMap<String, Float>> floatObservable= ForecastApi.Instance.getForecastApi().getWeatherDataByCoord(lat, lon, units).map(
+                new Function<WeatherEntity, HashMap<String, Float>>() {
+                    @Override
+                    public HashMap<String, Float> apply(@NonNull WeatherEntity weatherEntity) throws Exception {
+                        HashMap<String,Float> dailyForecast= new HashMap<>();
+                        dailyForecast.put(Constant.day, weatherEntity.getDaily()[0].getTemp().getDay());
+                        dailyForecast.put(Constant.eve, weatherEntity.getDaily()[0].getTemp().getEve());
+                        dailyForecast.put(Constant.morn, weatherEntity.getDaily()[0].getTemp().getMorn());
+                        dailyForecast.put(Constant.night, weatherEntity.getDaily()[0].getTemp().getNight());
+                        return dailyForecast;
```

Comment:
This logic should be inside mapper. Also it is not good idea to return map as domain object.

Reply:
I just create a HashMap to transfer it to ui

Reply:
I only need this once. For this I need to create a mapper?